ctk/idp/src/main/kotlin/org/codice/compliance/web/slo/RedirectSLOTest.kt

```diff
@@ -0,0 +1,138 @@
+                isSamlRequest = true
+            }.decodeAndVerify()
+            CoreLogoutRequestProtocolVerifier(samlLogoutRequestDom).verify()
+
+            val secondSPLogoutResponse = createDefaultLogoutResponse(samlLogoutRequestDom, true)
+            val encodedSecondSPLogoutResponse = encodeRedirectRequest(secondSPLogoutResponse)
+            val secondSPResponseQueryParams = SimpleSign().signUriString(
+                SAML_RESPONSE,
+                encodedSecondSPLogoutResponse,
+                null)
+            val logoutResponse = sendRedirectLogoutMessage(secondSPResponseQueryParams, cookies)
+
+            useDefaultServiceProvider()
+            val samlResponseDom = logoutResponse.getBindingVerifier().decodeAndVerify()
+            CoreLogoutResponseProtocolVerifier(logoutRequest, samlResponseDom,
+                logoutResponse.determineBinding()).verify()
+        }
+
+        "Redirect LogoutRequest Test With Relay State - Single SP" {
+            val cookies = loginAndGetCookies(HTTP_REDIRECT)
+
+            val logoutRequest = createDefaultLogoutRequest(HTTP_REDIRECT)
+            val encodedRequest = encodeRedirectRequest(logoutRequest)
+            val queryParams = SimpleSign().signUriString(
+                SAML_REQUEST,
+                encodedRequest,
+                EXAMPLE_RELAY_STATE)
+            val response = sendRedirectLogoutMessage(queryParams, cookies)
+
+            val samlResponseDom = response.getBindingVerifier().apply {
+                isRelayStateGiven = true
+            }.decodeAndVerify()
+            CoreLogoutResponseProtocolVerifier(logoutRequest, samlResponseDom,
+                response.determineBinding()).verify()
+        }
+
+        "Redirect LogoutRequest Test With Relay State - Multiple SPs" {
+            val cookies = loginAndGetCookies(HTTP_REDIRECT, singleSP = false)
+
+            val logoutRequest = createDefaultLogoutRequest(HTTP_REDIRECT)
+            val encodedRequest = encodeRedirectRequest(logoutRequest)
+            val queryParams = SimpleSign().signUriString(
+                SAML_REQUEST,
+                encodedRequest,
+                EXAMPLE_RELAY_STATE)
+            val secondSPLogoutRequest = sendRedirectLogoutMessage(queryParams, cookies)
+
+            useDSAServiceProvider()
+            val samlLogoutRequestDom = secondSPLogoutRequest.getBindingVerifier().apply {
+                isSamlRequest = true
+            }.decodeAndVerify()
+            CoreLogoutRequestProtocolVerifier(samlLogoutRequestDom).verify()
+
+            val secondSPLogoutResponse = createDefaultLogoutResponse(samlLogoutRequestDom, true)
+            val encodedSecondSPLogoutResponse = encodeRedirectRequest(secondSPLogoutResponse)
+            val secondSPResponseQueryParams = SimpleSign().signUriString(
+                SAML_RESPONSE,
+                encodedSecondSPLogoutResponse,
+                null)
```

Comment:
❗️ Shouldn't we be getting the relayState from the `secondSPLogoutRequest ` and passing that in here?